src/EditorFeatures/TestUtilities2/Intellisense/ModernCompletionTestState.vb

```diff
@@ -0,0 +1,639 @@
+                documentElement As XElement,
+                Optional extraCompletionProviders As CompletionProvider() = Nothing,
+                Optional excludedTypes As List(Of Type) = Nothing,
+                Optional extraExportedTypes As List(Of Type) = Nothing,
+                Optional includeFormatCommandHandler As Boolean = False) As ModernCompletionTestState
+            Return New ModernCompletionTestState(
+                <Workspace>
+                    <Project Language="C#" CommonReferences="true">
+                        <Document>
+                            <%= documentElement.Value %>
+                        </Document>
+                    </Project>
+                </Workspace>,
+                CreateLazyProviders(extraCompletionProviders, LanguageNames.CSharp, roles:=Nothing), ' Add in the Editor Completion mef components; Get the completion iolecommandtarget to send commands to and do normal assertions based on the old (/current) CompletionBroker
+                excludedTypes,
+                extraExportedTypes,
+                includeFormatCommandHandler)
+        End Function
+
+        Public Shared Function CreateTestStateFromWorkspace(
+                workspaceElement As XElement,
+                Optional extraCompletionProviders As CompletionProvider() = Nothing,
+                Optional extraExportedTypes As List(Of Type) = Nothing,
+                Optional workspaceKind As String = Nothing) As ModernCompletionTestState
+            Return New ModernCompletionTestState(
+                workspaceElement,
+                CreateLazyProviders(extraCompletionProviders, LanguageNames.VisualBasic, roles:=Nothing),
+                excludedTypes:=Nothing,
+                extraExportedTypes,
+                workspaceKind:=workspaceKind)
+        End Function
+
+#Region "IntelliSense Operations"
+
+        Public Overloads Sub SendEscape() Implements ITestState.SendEscape
+            Dim handler = DirectCast(EditorCompletionCommandHandler, VSCommanding.ICommandHandler(Of EscapeKeyCommandArgs))
+            MyBase.SendEscape(Sub(a, n, c) handler.ExecuteCommand(a, n, c), Sub() Return)
+        End Sub
+
+        Public Overloads Sub SendDownKey() Implements ITestState.SendDownKey
+            Throw New ArgumentException("Up Key test should be implemented on the Editor/VSSDK side")
+        End Sub
+
+        Public Overloads Sub SendUpKey() Implements ITestState.SendUpKey
+            Throw New ArgumentException("Up Key test should be implemented on the Editor/VSSDK side")
+        End Sub
+
+#End Region
+
+#Region "Completion Operations"
+        Public Overloads Sub SendTab() Implements ITestState.SendTab
+            Dim handler = DirectCast(EditorCompletionCommandHandler, VSCommanding.IChainedCommandHandler(Of TabKeyCommandArgs))
+            MyBase.SendTab(Sub(a, n, c) handler.ExecuteCommand(a, n, c), Sub() EditorOperations.InsertText(vbTab))
+        End Sub
```

Comment:
this all looks like duplication of the normal completion test state code... why do we need to duplciate things?  why not have them have a base class that shares as much as possible?